From: Peter Lemenkov <lemenkov@gmail.com>
Date: Fri, 9 Aug 2024 13:37:30 +0200
Subject: [PATCH] Fix test with self-signed certificates

* Add cacerts param
* Add verify_fun param
* Explicitly use ssl:handshake/1

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/src/riak_core_tcp_mon.erl b/src/riak_core_tcp_mon.erl
index 29e678f..101c04f 100644
--- a/src/riak_core_tcp_mon.erl
+++ b/src/riak_core_tcp_mon.erl
@@ -466,7 +466,7 @@ ssl_test_() ->
         spawn(fun () ->
             %% server
             {ok, S} = ssl:transport_accept(LS),
-            {ok, SslSock} = case ssl_handshake(S) of
+            {ok, SslSock} = case ssl:handshake(S) of
                 ok ->
                     {ok, S};
                 {ok, NewSocket} ->
@@ -477,7 +477,7 @@ ssl_test_() ->
             ssl_recv_loop(SslSock)
         end),
 
-        {ok, Socket} = ssl:connect("localhost", Port, [binary, {active, true}, {certfile, "test/site2-cert.pem"}, {keyfile, "test/site2-key.pem"}]),
+        {ok, Socket} = ssl:connect("localhost", Port, [binary, {active, true}, {cacerts, public_key:cacerts_get()}, {certfile, "test/site2-cert.pem"}, {keyfile, "test/site2-key.pem"}, {verify_fun, {fun(_,valid,U) -> {valid,U}; (_,valid_peer,U) -> {valid,U}; (_,{bad_cert, selfsigned_peer},U) -> {valid, U}; (_,{bad_cert, unknown_ca},U) -> {valid, U}; (_,Reason,_U) -> {fail, Reason} end, []}}]),
         riak_core_tcp_mon:monitor(Socket, "test", ssl),
         % so we have stats to see
         lists:foreach(fun(_) ->
